src/Control/Distributed/Process/Internal/Types.hs

```diff
@@ -137,7 +137,7 @@ import GHC.Generics
 newtype NodeId = NodeId { nodeAddress :: NT.EndPointAddress }
   deriving (Eq, Ord, Typeable, Data, Generic)
 instance Binary NodeId where
-instance NFData NodeId
+instance NFData NodeId where rnf (NodeId a) = rnf a `seq` ()
```

Comment:
There are 2 possibilities, use generic deriving on `>deepseq-1.4` and write instances manually on previous versions, or write instances manually everywhere. I find latter approach more straightforward as former implies latter under if-condition.